### PR TITLE
feat(website): ecosystem page

### PR DESCRIPTION
**Pull request by hadzhehsen**

### CODE REVIEW

File: libs/website/ecosystem-fund-page/src/lib/apply-block/apply-block.tsx [hunk 1]
```diff
@@ -24,7 +22,7 @@ export function ApplyBlock() {
         target="_blank"
         rel={'noopener noreferrer'}
       >
-        <Button variant={2} className="uppercase mt-[24px]">
+        <Button variant={2} className="mt-[24px]">
```
Comment (olegshilov) [inline libs/website/ecosystem-fund-page/src/lib/apply-block/apply-block.tsx]: wrap button with div and add magrin to this div

File: libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+type PartnerType =
+  | 'infrastructure'
+  | 'wallet'
+  | 'defi'
+  | 'bridge'
+  | 'payments'
+  | 'service';
+
+type PartnerStatus = 'live' | 'planned';
```
Comment (olegshilov) [inline libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx]: better to use enums here

File: libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+export interface ParterCardProps {
+  logo: ReactElement;
```
Comment (olegshilov) [inline libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx]:
```suggestion
  logo: ReactNode;
```

File: libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+        <div className="flex items-center ml-[48px]">
+          <div className="bg-haqq-black w-[16px] h-[16px]" />
+          <div className="h-[1px] bg-[#0d0d0e3d] w-[1200px] absolute" />
+        </div>
+      </div>
+
+      <Tabs className="mt-[28px] md:mt-[40px] lg:mt-[80px] overflow-x-auto md:overflow-x-hidden">
+        <Tab
+          isActive={tab === 'all-partners'}
+          onClick={() => setTab('all-partners')}
```
Comment (olegshilov) [inline libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx]:
```suggestion
          onClick={() => {
            setTab('all-partners')
          }
```

File: libs/website/ui-kit/src/lib/tabs/tabs.tsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+export function Tab({
+  children,
+  isActive = false,
+  onClick,
+}: PropsWithChildren<TabProps>) {
+  return (
+    <div
+      className={clsx(
+        'px-[16px] py-[12px]',
+        'font-sans text-[12px] leading-[1.5em] md:text-[13px] md:leading-[22px] lg:text-[14px] font-[500] ',
```
Comment (olegshilov) [inline libs/website/ui-kit/src/lib/tabs/tabs.tsx]:
```suggestion
        'font-sans text-[12px] leading-[1.5em] md:text-[13px] md:leading-[22px] lg:text-[14px] font-[500]',
```

**Review by olegshilov — CHANGES_REQUESTED** [review]

File: libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx (hunk 3 above)
Comment (hadzhehsen) [reply libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx]: Done

File: libs/website/ui-kit/src/lib/tabs/tabs.tsx (hunk 5 above)
Comment (hadzhehsen) [reply libs/website/ui-kit/src/lib/tabs/tabs.tsx]: Done

File: libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx (hunk 4 above)
Comment (hadzhehsen) [reply libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx]: Done

File: libs/website/ecosystem-fund-page/src/lib/apply-block/apply-block.tsx (hunk 1 above)
Comment (hadzhehsen) [reply libs/website/ecosystem-fund-page/src/lib/apply-block/apply-block.tsx]: Done

Comment (hadzhehsen) [reply libs/website/ecosystem-fund-page/src/lib/apply-block/apply-block.tsx]: Done

File: libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx (hunk 2 above)
Comment (hadzhehsen) [reply libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx]: Done

File: libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx (hunk 3 above)
Comment (hadzhehsen) [reply libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx]: Done

File: libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx (hunk 4 above)
Comment (hadzhehsen) [reply libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx]: Done

File: libs/website/ui-kit/src/lib/tabs/tabs.tsx (hunk 5 above)
Comment (hadzhehsen) [reply libs/website/ui-kit/src/lib/tabs/tabs.tsx:35]: Done

File: libs/website/ecosystem-page/src/lib/apply-block/apply-block.tsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
+import { Button, Heading, Text } from '@haqq/website/ui-kit';
+import Image from 'next/image';
+import Link from 'next/link';
+import sunshineBgData from '../../assets/sunshine.svg';
+
+export function ApplyBlock() {
+  return (
+    <section className="border-t border-t-[#2A2A2B] overflow-hidden relative pt-[68px] pb-[80px] px-[16px] sm:px-[48px] lg:px-[80px]">
+      <div className="relative z-10 flex flex-col">
+        <Text className="text-haqq-gold"> Grants and Investment Program</Text>
```
Comment (olegshilov) [inline libs/website/ecosystem-page/src/lib/apply-block/apply-block.tsx]:
```suggestion
        <Text className="text-haqq-gold">Grants and Investment Program</Text>
```

File: libs/website/ecosystem-page/src/lib/apply-block/apply-block.tsx [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+        <Heading className="mt-[8px]">
+          Apply now for the Haqq Ecosystem Fund, and let's{' '}
```
Comment (olegshilov) [inline libs/website/ecosystem-page/src/lib/apply-block/apply-block.tsx]:
```suggestion
          Apply now for the Haqq Ecosystem Fund, and let's
```

File: apps/website/pages/ecosystem.tsx [hunk 8]
```diff
@@ -0,0 +1,32 @@
+import { getStoryblokApi } from '@storyblok/react';
+import { storyblokInit, apiPlugin } from '@storyblok/react';
```
Comment (olegshilov) [inline apps/website/pages/ecosystem.tsx]:
```suggestion
import { getStoryblokApi, storyblokInit, apiPlugin } from '@storyblok/react';
```

File: libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx [hunk 9]
```diff
@@ -0,0 +1,268 @@
...
+      case PartnerType.PAYMENTS:
+        text = 'Payments';
+        break;
+      case PartnerType.SERVICE:
+        text = 'Service';
+        break;
+      default:
+        break;
+    }
+  }
```
Comment (olegshilov) [inline libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx]:
```suggestion
  const text = useMemo(() => {
    if (status) {
      switch (status) {
        case PartnerStatus.LIVE:
          return "Live";
        case PartnerStatus.PLANNED:
          return "Planned";

        default:
          break;
      }
    } else if (type) {
      switch (type) {
        case PartnerType.INFRASTRUCTURE:
          return "Infrastructure";

        case PartnerType.WALLET:
          return "Wallet";

        case PartnerType.DEFI:
          return "DeFi";

        case PartnerType.BRIDGE:
          return "Bridge";

        case PartnerType.PAYMENTS:
          return "Payments";

        case PartnerType.SERVICE:
          return "Service";
      }
    } else {
      return undefined;
    }
  }, []);
```

File: libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx [hunk 10]
```diff
@@ -0,0 +1,268 @@
...
+    type: PartnerType | 'all-partners',
+  ) => {
+    if (type === 'all-partners') {
+      return partners;
+    }
+
+    return partners.filter((partner) => partner.type === type);
+  };
+
+  const filteredPartners = filterPartnersByType(partners, tab);
```
Comment (olegshilov) [inline libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx]: use useMemo hook here

File: libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx [hunk 11]
```diff
@@ -0,0 +1,268 @@
...
+
+  const imageDimensions = (imageUrl: string) => {
```
Comment (olegshilov) [inline libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx]: function or useCallback here

File: libs/website/index-page/src/assets/images/ecliplse.png
Comment (olegshilov) [inline libs/website/index-page/src/assets/images/ecliplse.png:1]: rename file to eclipse.png

File: libs/website/ui-kit/src/lib/tabs/tabs.tsx (hunk 5 above)
Comment (olegshilov) [reply libs/website/ui-kit/src/lib/tabs/tabs.tsx]: not done

**Review by olegshilov — CHANGES_REQUESTED** [review]

File: libs/website/ui-kit/src/lib/tabs/tabs.tsx (hunk 5 above)
Comment (hadzhehsen) [reply libs/website/ui-kit/src/lib/tabs/tabs.tsx]: Resolved

File: libs/website/ecosystem-page/src/lib/apply-block/apply-block.tsx (hunk 6 above)
Comment (hadzhehsen) [reply libs/website/ecosystem-page/src/lib/apply-block/apply-block.tsx]: Resolved

File: libs/website/ecosystem-page/src/lib/apply-block/apply-block.tsx (hunk 7 above)
Comment (hadzhehsen) [reply libs/website/ecosystem-page/src/lib/apply-block/apply-block.tsx]: Resolved

File: apps/website/pages/ecosystem.tsx (hunk 8 above)
Comment (hadzhehsen) [reply apps/website/pages/ecosystem.tsx]: Resolved

File: libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx (hunk 9 above)
Comment (hadzhehsen) [reply libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx]: Resolved

File: libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx (hunk 10 above)
Comment (hadzhehsen) [reply libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx]: Resolved

File: libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx (hunk 11 above)
Comment (hadzhehsen) [reply libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx]: Resolved

File: libs/website/index-page/src/assets/images/ecliplse.png
Comment (hadzhehsen) [reply libs/website/index-page/src/assets/images/ecliplse.png:1]: Resolved

File: libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx [hunk 12]
```diff
@@ -0,0 +1,262 @@
+import { useCallback, useMemo, useState } from 'react';
+import Link from 'next/link';
+import Image from 'next/image';
+import clsx from 'clsx';
+import { Heading, Tabs, Tab } from '@haqq/website/ui-kit';
+
+enum PartnerType {
```
Comment (olegshilov) [inline libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx]: export this

Comment (olegshilov) [inline libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx]: export this

File: apps/website/pages/ecosystem.tsx [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+      version: process.env.VERCEL_ENV === 'production' ? 'published' : 'draft',
+    });
+    data = response?.data;
+  } catch (error) {
+    console.error(error);
+  }
+
+  return {
+    props: {
+      story: data?.story ?? false,
```
Comment (olegshilov) [inline apps/website/pages/ecosystem.tsx]: return array with partners here

File: libs/website/ecosystem-page/src/lib/ecosystem-page/ecosystem-page.tsx [hunk 14]
```diff
@@ -0,0 +1,20 @@
+import { Fragment } from 'react';
+import Head from 'next/head';
+import { TitleBlock } from '../title-block/title-block';
+import { ApplyBlock } from '../apply-block/apply-block';
+import { PartnersBlock } from '../partners-block/partners-block';
+
+export function EcosystemPage({ story }) {
```
Comment (olegshilov) [inline libs/website/ecosystem-page/src/lib/ecosystem-page/ecosystem-page.tsx]: types missing

File: libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx [hunk 15]
```diff
@@ -0,0 +1,262 @@
...
+  INFRASTRUCTURE = 'infrastructure',
+  WALLET = 'wallet',
+  DEFI = 'defi',
+  BRIDGE = 'bridge',
+  PAYMENTS = 'payments',
+  SERVICE = 'service',
+}
+
+enum PartnerStatus {
```
Comment (olegshilov) [inline libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx]: export

File: libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx [hunk 16]
```diff
@@ -0,0 +1,262 @@
...
+  LIVE = 'live',
+  PLANNED = 'planned',
+}
+
+export interface Partner {
+  logo: {
+    filename: string;
+  };
```
Comment (olegshilov) [inline libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx]:
```suggestion
  logoUrl: string;
```

File: libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx [hunk 17]
```diff
@@ -0,0 +1,262 @@
...
+    if (type === 'all-partners') {
+      return partners;
+    }
+
+    return partners.filter((partner) => partner.type === type);
+  };
+
+  const filteredPartners = useMemo(() => {
+    return filterPartnersByType(partners, tab);
+  }, [partners, tab]);
```
Comment (olegshilov) [inline libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx]:
```suggestion
  const filteredPartners = useMemo(() => {
    if (type === "all-partners") {
      return partners;
    }

    return partners.filter((partner) => {
      return partner.type === type;
    });
  }, [partners, type]);
```

**Review by olegshilov — CHANGES_REQUESTED** [review]

**Review by olegshilov — CHANGES_REQUESTED** [review]

File: libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx (hunk 12 above)
Comment (hadzhehsen) [reply libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx]: Resolved

File: libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx (hunk 15 above)
Comment (hadzhehsen) [reply libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx]: Resolved

File: libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx (hunk 12 above)
Comment (hadzhehsen) [reply libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx]: Resolved

File: apps/website/pages/ecosystem.tsx [hunk 18]
```diff
@@ -0,0 +1,31 @@
...
+export { EcosystemPage as default } from '@haqq/website/ecosystem-page';
+
+export async function getStaticProps() {
+  let data;
+  const slug = 'partners';
+
+  storyblokInit({
+    accessToken: process.env['STORYBLOK_ACCESS_TOKEN'],
+    use: [apiPlugin],
+  });
```
Comment (olegshilov) [inline apps/website/pages/ecosystem.tsx]: mve storyblock init outside `getStaticProps` method

**Review by olegshilov — CHANGES_REQUESTED** [review]

File: apps/website/pages/ecosystem.tsx (hunk 13 above)
Comment (hadzhehsen) [reply apps/website/pages/ecosystem.tsx]: Resolved

File: libs/website/ecosystem-page/src/lib/ecosystem-page/ecosystem-page.tsx (hunk 14 above)
Comment (hadzhehsen) [reply libs/website/ecosystem-page/src/lib/ecosystem-page/ecosystem-page.tsx]: Resolved

File: libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx (hunk 16 above)
Comment (hadzhehsen) [reply libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx]: Resolved

File: libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx (hunk 17 above)
Comment (hadzhehsen) [reply libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx]: Resolved

File: apps/website/pages/ecosystem.tsx (hunk 18 above)
Comment (hadzhehsen) [reply apps/website/pages/ecosystem.tsx]: Resolved

File: libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx (hunk 15 above)
Comment (olegshilov) [reply libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx]: where?

File: libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx (hunk 16 above)
Comment (olegshilov) [reply libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx]: where?

File: libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx (hunk 17 above)
Comment (olegshilov) [reply libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx]: where?

File: libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx (hunk 12 above)
Comment (olegshilov) [reply libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx]: where?

Comment (olegshilov) [reply libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx]: where?

File: libs/website/ecosystem-page/src/lib/ecosystem-page/ecosystem-page.tsx (hunk 14 above)
Comment (olegshilov) [reply libs/website/ecosystem-page/src/lib/ecosystem-page/ecosystem-page.tsx]: where?

**Review by olegshilov — CHANGES_REQUESTED** [review]

File: apps/website/pages/ecosystem.tsx [hunk 19]
```diff
@@ -0,0 +1,47 @@
...
+        _uid: el._uid,
+        name: el.name,
+        logoUrl: el.logo.filename,
+        link: el.link,
+        description: el.description,
+        type: el.type,
+        status: el.status,
+      };
+    });
+  };
```
Comment (olegshilov) [inline apps/website/pages/ecosystem.tsx]: move this method outside `getStaticProps` and use function

File: apps/website/pages/ecosystem.tsx [hunk 20]
```diff
@@ -0,0 +1,47 @@
...
+
+  const partners = await mapStoryblockDataToPartners(data);
+
+  return {
+    props: {
+      partners: partners ?? false,
```
Comment (olegshilov) [inline apps/website/pages/ecosystem.tsx]:
```suggestion
      partners: partners ?? [],
```

**Review by olegshilov — CHANGES_REQUESTED** [review]

File: apps/website/pages/ecosystem.tsx (hunk 19 above)
Comment (olegshilov) [reply apps/website/pages/ecosystem.tsx]: and why it's async?

Comment (hadzhehsen) [reply apps/website/pages/ecosystem.tsx]: Fixed

File: apps/website/pages/ecosystem.tsx (hunk 20 above)
Comment (hadzhehsen) [reply apps/website/pages/ecosystem.tsx]: Fixed

File: apps/website/pages/ecosystem.tsx [hunk 21]
```diff
@@ -0,0 +1,47 @@
+import { getStoryblokApi, storyblokInit, apiPlugin } from '@storyblok/react';
+
+storyblokInit({
+  accessToken: process.env.STORYBLOK_ACCESS_TOKEN,
+  use: [apiPlugin],
+});
+
+function mapStoryblockDataToPartners(data) {
```
Comment (olegshilov) [inline apps/website/pages/ecosystem.tsx]: types missing

File: apps/website/pages/ecosystem.tsx [hunk 22]
```diff
@@ -0,0 +1,47 @@
...
+    const storyblokApi = getStoryblokApi();
+    const response = await storyblokApi.get(`cdn/stories/${slug}`, {
+      version: process.env.VERCEL_ENV === 'production' ? 'published' : 'draft',
+    });
+    data = response?.data;
+  } catch (error) {
+    console.error(error);
+  }
+
+  const partners = await mapStoryblockDataToPartners(data);
```
Comment (olegshilov) [inline apps/website/pages/ecosystem.tsx]: function is not async

**Review by olegshilov — CHANGES_REQUESTED** [review]

File: apps/website/pages/ecosystem.tsx (hunk 22 above)
Comment (hadzhehsen) [reply apps/website/pages/ecosystem.tsx]: Resolved

File: libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx [hunk 23]
```diff
@@ -0,0 +1,263 @@
...
+}
+
+export function PartnersBlock({ partners }: { partners: Partner[] }) {
+  const [tab, setTab] = useState<PartnerType | 'all-partners'>('all-partners');
+
+  const filteredPartners = useMemo(() => {
+    if (tab === 'all-partners') {
+      return partners;
+    }
+    return partners.filter((partner) => partner.type === tab);
```
Comment (olegshilov) [inline libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx]:
```suggestion
    return partners.filter((partner) => {
      return partner.type === tab;
    });
```

File: libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx [hunk 24]
```diff
@@ -0,0 +1,263 @@
...
+            type={partner.type}
+            status={partner.status}
+            name={partner.name}
+            logoUrl={partner.logoUrl}
+            link={partner.link}
+            description={partner.description}
+            logoHeight={imageDimensions(partner.logoUrl).height}
+            logoWidth={imageDimensions(partner.logoUrl).width}
+          />
+        ))}
```
Comment (olegshilov) [inline libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx]:
```suggestion
        {filteredPartners?.map((partner: Partner) => {
          return (
            <PartnerCard
              key={partner._uid}
              type={partner.type}
              status={partner.status}
              name={partner.name}
              logoUrl={partner.logoUrl}
              link={partner.link}
              description={partner.description}
              logoHeight={imageDimensions(partner.logoUrl).height}
              logoWidth={imageDimensions(partner.logoUrl).width}
            />
          )
        })}
```

**Review by olegshilov — CHANGES_REQUESTED** [review]

File: libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx (hunk 24 above)
Comment (hadzhehsen) [reply libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx]: Resolved

File: libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx (hunk 23 above)
Comment (hadzhehsen) [reply libs/website/ecosystem-page/src/lib/partners-block/partners-block.tsx]: Resolved

File: apps/website/pages/ecosystem.tsx (hunk 21 above)
Comment (hadzhehsen) [reply apps/website/pages/ecosystem.tsx]: Resolved

**Review by kioqq — APPROVED** [review]

**Review by kioqq — APPROVED** [review]

**Review by olegshilov — APPROVED** [review]

**Pull request merged**